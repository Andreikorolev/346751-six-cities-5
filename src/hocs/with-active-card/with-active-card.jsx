import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {compose} from "redux";
import {connect} from "react-redux";
import {ActionCreator} from "../../store/action";

const withActiveCard = (Component) => {
  class WithActiveCard extends PureComponent {
    constructor(props) {
      super(props);

      this.state = {
        activeCard: -1,
      };
    }

    render() {
      const {onCardMouseOver} = this.props;

      return (
        <Component
          {...this.props}
          onCardMouseOver={(cardId) => {
            this.setState(() => ({
              activeCard: cardId,
            }));
            onCardMouseOver(cardId);
          }}
        />
      );
    }
  }

  WithActiveCard.propTypes = {
    onCardMouseOver: PropTypes.func.isRequired,
  };

  return WithActiveCard;
};


const mapStateToProps = (state) => ({
  activeCard: state.activeCard,
});

const mapDispatchToProps = (dispatch) => ({
  onCardMouseOver(cardId) {
    dispatch(ActionCreator.activeCardChange(cardId));
  },
});

export {withActiveCard};
export default compose(connect(mapStateToProps, mapDispatchToProps), withActiveCard);
