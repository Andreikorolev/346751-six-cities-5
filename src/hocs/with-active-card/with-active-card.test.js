import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import {withActiveCard} from "./with-active-card";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

const MockComponentHOC = withActiveCard(MockComponent);

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

describe(`withActiveCard render`, () => {
  it(`shoul render correctly`, () => {
    const tree = renderer.create(
        <MockComponentHOC
          onCardMouseOver={() => {}}
        >
          <React.Fragment/>
        </MockComponentHOC>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
