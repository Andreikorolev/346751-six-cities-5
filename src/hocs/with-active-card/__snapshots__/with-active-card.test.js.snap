// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withActiveCard render shoul render correctly 1`] = `<div />`;
